app/models/ability.rb

```diff
@@ -0,0 +1,40 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+
+    if user.is? :admin
+      can :manage, :all
+    else
+      can :manage, Post, author_id: user.id
+      can :manage, Comment, author_id: user.id
+      can :read, :all
```

Comment:
I think we don't need a role as per the requirements. You can get rid of the role